### PR TITLE
Add makefile and targetted runs

### PR DESCRIPTION
**Pull request by owenrumney**
Just a selfish convenience really, won't be offended if you opt to reject

### CODE REVIEW

File: run_all_tools.sh [hunk 1]
```diff
@@ -1,4 +1,61 @@
...
+
+# Cloudrail
+function run_cloudrail {
+  echo Now running Cloudrail on all cases
+  docker pull indeni/cloudrail-cli:latest
+  docker run -t -v $PWD:/tf indeni/cloudrail-cli --version | awk '{print $NF}' > version_cloudrail.txt
+  find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do echo $test_case ; ORG_PATH=$PWD ; cd $test_case ; if [ ! -f cloudrail_results.txt ]; then docker run --rm -v $PWD:/data -e CLOUDRAIL_API_KEY indeni/cloudrail-cli run --tf-plan plan.out --output-file cloudrail_results.txt --no-cloud-account --auto-approve -v; fi ; cd $ORG_PATH; done
+}
+
+case $1 in
```
Comment (yi2020) [inline run_all_tools.sh:51]: This needs to be moved to the bottom. Some of the tools require the plan.out files that are generated further below.

Also, run_snyk requires the SNYK_TOKEN, and run_cloudrail requires the CLOUDRAIL_API_KEY. You could move those if's into their respective functions.

File: run_all_tools.sh [hunk 2]
```diff
@@ -24,38 +81,9 @@ export AWS_REGION=us-west-1
...
-docker pull snyk/snyk-cli:docker
-docker run -t -v empty:/project -e SNYK_TOKEN snyk/snyk-cli:docker --version | tail -n 1 > version_snyk.txt
-find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do echo $test_case ; ORG_PATH=$PWD ; cd $test_case ; if [ ! -f snyk_results.txt ]; then docker run --rm -v "$(pwd):/project" -e SNYK_TOKEN snyk/snyk-cli:docker iac test --experimental /project | sed "s~$ORG_PATH~tool-compare~" > snyk_results.txt ; fi ; cd $ORG_PATH; done
-
-# Cloudrail
-echo Now running Cloudrail on all cases
-docker pull indeni/cloudrail-cli:latest
-docker run -t -v $PWD:/tf indeni/cloudrail-cli --version | awk '{print $NF}' > version_cloudrail.txt
-find . -name "main.tf" -exec dirname {} \; | grep -v "\.terraform" | while read -r test_case; do echo $test_case ; ORG_PATH=$PWD ; cd $test_case ; if [ ! -f cloudrail_results.txt ]; then docker run --rm -v $PWD:/data -e CLOUDRAIL_API_KEY indeni/cloudrail-cli run --tf-plan plan.out --output-file cloudrail_results.txt --no-cloud-account --auto-approve -v; fi ; cd $ORG_PATH; done
+run_checkov
```
Comment (yi2020) [inline run_all_tools.sh:84]: The code structure is a bit odd. how about adding the calling of all functions as a default in the `case` statement you have? (use `*)`)

**Review by yi2020 — COMMENTED** [review]
I like this @owenrumney and would use it myself, nice work :) Added a couple of comments.

File: run_all_tools.sh (hunk 1 above)
Comment (owenrumney) [reply run_all_tools.sh]: Yup, that makes sense - didn't appreciate the relevance of the plan generation. Fixed

**Comment by yi2020** [comment]
@owenrumney also note this branch now has a conflict due to a separate PR that I merged.

**Comment by owenrumney** [comment]
No probs, I'm doing a full run of all tests so I've pull from upstream. Each individual works, just testing the full end to end then will push

**Pull request merged**